toolkit/scripts/tools.mk

```diff
@@ -116,6 +116,7 @@ $(TOOL_BINS_DIR)/%: $(go_common_files)
 		go test -ldflags="$(go_ldflags)" -test.short -covermode=atomic -coverprofile=$(BUILD_DIR)/tools/$*.test_coverage ./... && \
 		CGO_ENABLED=0 go build \
 			-ldflags="$(go_ldflags)" \
+			$(if $(filter y,$(BUILD_TOOLS_NONPROD)),,-tags prod) \
```

Comment:
What is this change about? `BUILD_TOOLS_NONPROD` looks like a new variable and it's not defined or documented anywhere.

Reply:
I'm using it to create variant behavior of the code depending on whether or not the `prod` build tag is specified or not. See, [passwordtype_prod.go](https://github.com/microsoft/azurelinux/pull/9070/files#diff-94f99db37ee8590f52d2abae800a4bbb72030d1a6ec03302f6e1fed2cb8bb703R1) and [passwordtype_nonprod.go
](https://github.com/microsoft/azurelinux/pull/9070/files#diff-690b124ac4d99d31a420c5ce2b50d09b135500f4ed1352b377164b7c4fa8b905R1).

`BUILD_TOOLS_NONPROD` is used to control whether or not the tag is specified.

Reply:
Actually, I might look into how easy it would be to only omit the `prod` build tag for the `osmodifier` tool.

Reply:
I read the rest of the code and I like the idea but please add some documentation for `BUILD_TOOLS_NONPROD`.

Reply:
Fixed.

Reply:
I just refreshed the page and saw your two replies to my first comment (I added my second one thinking there was no reply, yet). If you still want to work without the `prod` tag, then I've got nothing against that as well.